src/plugins/intel_cpu/src/nodes/executors/transpose_list.hpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "executor.hpp"
+
+#include "transpose.hpp"
+#if defined(OV_CPU_WITH_ACL)
+#include "acl/acl_transpose.hpp"
+#endif
+
+#include "common/ref_transpose.hpp"
+#include "dnnl/dnnl_transpose.hpp"
+
+#include "onednn/iml_type_mapper.h"
+#include "common/primitive_cache.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+struct TransposeExecutorDesc {
+    ExecutorType executorType;
+    TransposeExecutorBuilderCPtr builder;
+};
+
+const std::vector<TransposeExecutorDesc>& getTransposeExecutorsList();
+
+class TransposeExecutorFactory : public ExecutorFactory {
+public:
+TransposeExecutorFactory(const TransposeParams& transposeParams,
+                         const std::vector<MemoryDescPtr>& srcDescs,
+                         const std::vector<MemoryDescPtr>& dstDescs,
+                         const ExecutorContext::CPtr context) : ExecutorFactory(context) {
+    for (auto& desc : getTransposeExecutorsList()) {
+        if (desc.builder->isSupported(transposeParams, srcDescs, dstDescs)) {
+            supportedDescs.push_back(desc);
+        }
+    }
+}
+
+~TransposeExecutorFactory() = default;
+virtual TransposeExecutorPtr makeExecutor(const TransposeParams& transposeParams,
```

Comment:
There is little sense to keep this method in the header, being called indirectly it will not be inline until the specific type is used. And, there is not many advantages of inlining this code, mostly because it contains another calls internally.

Reply:
fixed